.gitlab-ci.yml

```diff
@@ -81,6 +87,26 @@ clippy:
 
 #### stage:                        test (all features)
 
+test-dylint:
+  stage:                           test
+  <<:                              *docker-env
+  script:
+    - cd ink_linting/
+
+    # Installing these components here is necessary because
+    # `ink_linting/` has a fixed `rust-toolchain` file.
+    # We can't move this line to the Docker container, since
+    # that would then make it impossible to upgrade the
+    # `ink_linting/rust-toolchain` file while still having
+    # this CI job succeed.
+    - rustup component add rustfmt clippy rust-src
+
+    - cargo check --verbose
+    - cargo fmt --verbose --all -- --check
+    - cargo clippy --verbose -- -D warnings;
+
+    - cargo test --verbose --workspace --all-features
```

Comment:
Is the `--workspace` necessary? Seems to run without it. Harmless anyway

Reply:
Yes. Because we are not using a virtual manifest. All commands only run on the main package in thise case without `--workspace` or a `default-member` key in the manifest.